.github/workflows/compose.yml

```diff
@@ -0,0 +1,57 @@
+name: Docker Compose deployment 
+
+on:
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  docker-compose:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Log into Docker registry
+      env:
+        DOCKER_REGISTRY_URL: tip-tip-wlan-cloud-ucentral.jfrog.io
+        DOCKER_REGISTRY_USERNAME: ucentral
+      uses: docker/login-action@v1
+      with:
+        registry: ${{ env.DOCKER_REGISTRY_URL }}
+        username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWORD }}
+
+    - name: Create certificate and key files from Github secrets
+      env: 
+        DIGICERT_TIP_WILDCARD_CERT: ${{ secrets.DIGICERT_TIP_WILDCARD_CERT }}
+        DIGICERT_TIP_WILDCARD_KEY: ${{ secrets.DIGICERT_TIP_WILDCARD_KEY }}
+      run: |
+        cp certificates/root.pem docker-compose/ucentral-data/certs/restapi-ca.pem
+        'echo "$DIGICERT_TIP_WILDCARD_CERT" > docker-compose/ucentral-data/certs/{websocket-cert.pem,restapi-cert.pem}'
+        'echo "$DIGICERT_TIP_WILDCARD_KEY" > docker-compose/ucentral-data/certs/{websocket-key.pem,restapi-key.pem}'
+
+    - name: Instantiate Docker Compose deployment
+      env:
+        DOCKER_COMPOSE_PROJECT_NAME: ucentral-${{ github.sha }}
+        UCENTRALGW_TAG: ${{ github.sha }}      
```

Comment:
Are you sure that the uCentral GW image with this tag is available at this step? I know that there's a workflow responsible for building the image but both workflows will be triggered at the same time and I don't think the build will be done by then.

Reply:
No unfortunately it isn't available at this step. I think I'll put the deployment in the same workflow as the Docker build and set a condition that it is only triggered on pull request events.

Reply:
:+1: 
Honestly I think we can even keep it enabled for pushes to master.